src/VisualStudio/Core/Def/ExternalAccess/VSTypeScript/Api/VSTypeScriptVisualStudioProjectWrapper.LSPContainedDocumentServiceProvider.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.VisualStudio.LanguageServices.ExternalAccess.VSTypeScript.Api
+{
+    internal sealed partial class VSTypeScriptVisualStudioProjectWrapper
+    {
+        private sealed class LspContainedDocumentServiceProvider : IDocumentServiceProvider, IDocumentOperationService
+        {
+            private readonly VirtualDocumentPropertiesService _documentPropertiesService;
+
+            private LspContainedDocumentServiceProvider()
+            {
+                _documentPropertiesService = VirtualDocumentPropertiesService.Instance;
+            }
+
+            public static LspContainedDocumentServiceProvider Instance = new LspContainedDocumentServiceProvider();
+
+            bool IDocumentOperationService.CanApplyChange => true;
+
+            bool IDocumentOperationService.SupportDiagnostics => true;
+
+            TService? IDocumentServiceProvider.GetService<TService>() where TService : class
+            {
+                if (typeof(TService) == typeof(DocumentPropertiesService))
+                {
+                    return (TService)(object)_documentPropertiesService;
+                }
+
+                return this as TService;
+            }
+
+            private sealed class VirtualDocumentPropertiesService : DocumentPropertiesService
+            {
+                private const string _lspClientName = "TypeScript";
```

Comment:
Is your use of LspClientName here new?

Reply:
Yes, it is. This is to keep documents that correspond to script block contents (i.e. "virtual documents") from having their diagnostics published to the error list table control in a manner other than through the lsp publish diagnostics notifications, as Razor does with their generated .cs files.

Reply:
Alright just going to tag @CyrusNajmabadi and @dibarbet so they know that's happening and it's not just Razor using this mechanism anymore.